### ISSUE TITLE
feat[dev]: ✨ implement A11y Re-Improvements to sd-tooltip

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#2064]
## Description
This tasks describes the Figma implementation for: #2048 

This task describes the Storybook implementation for the sd-tooltip component as part of the A11y re-improvements outlined in Epic #2048.

A11y improvements for sd-tooltip can be found here: [sd-tooltip Notes](https://www.notion.so/matuzo/sd-tooltip-1baf3f6e1bc081f186edd618c62683d7).

------------------------------------------

# New Comment
**Problem:** The button doesn’t indicate whether it’s open.
**Solution:** Put `aria-expanded` on the button and toggle it on click.

---

**Problem:** Focus is lost when you click a button. 
**Solution:** Keep focus on the button.

---

**Problem:** Buttons are not accessible with VoiceOVer in Safari 18.2 on macOS.
**Solution:** I'm not exactly sure what the reason is but it may have to do with you using `display: contents`.

## Comment
/

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
